semgrep_output_v1.atd

```diff
@@ -2242,10 +2242,19 @@ type resolution_cmd_failed
  * lockfile parsing might partially fail but recover and still produce results)
 *)
 type resolution_result = [
-  | ResolutionOk of (found_dependency list * resolution_error list)
+  | ResolutionOk of (sca_dependency list * resolution_error list)
   | ResolutionError of resolution_error list
 ]
 
+(* A found_dependency annotated with extra information that is used
+ * within the Semgrep CLI, passed back and forth from OCaml to Python
+ * via RPC. See also SCA_dependency.t in OCaml and ResolvedDependencies 
+ * in Python.*)
+type sca_dependency = {
+  inherit found_dependency;
```

Comment:
I would avoid inherit. It's not super practice on the OCaml side because
you can't reuse record easily and for example build an sca_dependency from a found_dependency if
you used inherit because the generated OCaml code will generate 2 separate records.

We used inherit in the past because we had to to remain backward compatible with the generated JSON.

Reply:
Updated to use a separate record for the internal info that we want to track for downloaded dependencies